recon_linux.go

```diff
@@ -4,67 +4,69 @@ import (
 	"fmt"
 	"net"
 
-	"github.com/mudler/sendfd"
+	"github.com/ftrvxmtrx/fd"
 )
 
 // StartUnixListener starts up a unix domain socket listener for reconnects to
 // be sent through
-func (irc *Bot) StartUnixListener() {
-	unaddr, err := net.ResolveUnixAddr("unix", irc.unixastr)
+func (bot *Bot) StartUnixListener() {
+	unaddr, err := net.ResolveUnixAddr("unix", bot.unixastr)
 	if err != nil {
 		panic(err)
 	}
+
 	list, err := net.ListenUnix("unix", unaddr)
 	if err != nil {
 		panic(err)
 	}
+	defer list.Close()
+	bot.unixlist = list
 
-	irc.unixlist = list
 	con, err := list.AcceptUnix()
 	if err != nil {
 		fmt.Println("unix listener error: ", err)
 		return
 	}
-	list.Close()
+	defer con.Close()
 
-	fi, err := irc.con.(*net.TCPConn).File()
+	fi, err := bot.con.(*net.TCPConn).File()
 	if err != nil {
 		panic(err)
 	}
 
-	err = sendfd.SendFD(con, fi)
+	err = fd.Put(con, fi)
 	if err != nil {
 		panic(err)
 	}
 
 	select {
-	case <-irc.Incoming:
+	case <-bot.Incoming:
 	default:
-		close(irc.Incoming)
+		close(bot.Incoming)
 	}
-	close(irc.outgoing)
+	close(bot.outgoing)
 }
 
 // Attempt to hijack session previously running bot
-func (irc *Bot) hijackSession() bool {
-	unaddr, err := net.ResolveUnixAddr("unix", irc.unixastr) // The only way to get an error here is if the first parameter is not one of "unix", "unixgram" or "unixpacket". That will never happen.
-	if err != nil {
-		panic(err)
-	}
-	con, err := net.DialUnix("unix", nil, unaddr)
+func (bot *Bot) hijackSession() bool {
+	con, err := net.Dial("unix", bot.unixastr)
```

Comment:
If you use [net.DialUnix](https://golang.org/pkg/net/#DialUnix)  you can remove the type assertion further down.

Reply:
net.DialUnix can't use a string path it needs net.*UnixAddr which you can only get from net.ResolveUnixAddr, I think.

I find a simple dial and type assertion cleaner

Reply:
Ah, that makes sense.